Flaky test about BLMPOP in go cases

### Search before asking

- [x] I had searched in the [issues](https://github.com/apache/kvrocks/issues) and found no similar issues.


### Version

unstable

### Minimal reproduce step

https://github.com/apache/kvrocks/actions/runs/13037715492/job/36389804631?pr=2750#step:17:137
```
--- FAIL: TestList (22.59s)
    --- FAIL: TestList/BLMPOP_test_blocked_served_bothKey_FIFO_countOver_RIGHT#01 (0.10s)
        tcp_client.go:73: 
            	Error Trace:	/home/runner/work/kvrocks/kvrocks/tests/gocase/util/tcp_client.go:73
            	            				/home/runner/work/kvrocks/kvrocks/tests/gocase/util/tcp_client.go:100
            	            				/home/runner/work/kvrocks/kvrocks/tests/gocase/unit/type/list/list_test.go:1470
            	Error:      	Not equal: 
            	            	expected: "blmpop-list2"
            	            	actual  : "blmpop-list1"
            	            	
            	            	Diff:
            	            	--- Expected
            	            	+++ Actual
            	            	@@ -1 +1 @@
            	            	-blmpop-list2
            	            	+blmpop-list1
            	Test:       	TestList/BLMPOP_test_blocked_served_bothKey_FIFO_countOver_RIGHT#01
FAIL
```

### What did you expect to see?

--

### What did you see instead?

--

### Anything Else?

_No response_

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

--- a/tests/gocase/unit/type/list/list_test.go
+++ b/tests/gocase/unit/type/list/list_test.go
@@ -1462,15 +1462,16 @@ func testList(t *testing.T, configs util.KvrocksServerConfigs) {
 			// https://github.com/apache/kvrocks/issues/2617
 			// WriteArgs are required to be executed first
 			time.Sleep(100 * time.Millisecond)
-			require.NoError(t, rdb.RPush(ctx, key2, "one", "two").Err())
+
 			require.NoError(t, rdb.RPush(ctx, key1, "ONE", "TWO").Err())
+			require.NoError(t, rdb.RPush(ctx, key2, "one", "two").Err())
 			if direction == "LEFT" {
-				rd.MustReadStringsWithKey(t, key2, []string{"one", "two"})
+				rd.MustReadStringsWithKey(t, key1, []string{"ONE", "TWO"})
 			} else {
-				rd.MustReadStringsWithKey(t, key2, []string{"two", "one"})
+				rd.MustReadStringsWithKey(t, key1, []string{"TWO", "ONE"})
 			}
-			require.EqualValues(t, 0, rdb.Exists(ctx, key2).Val())
-			require.EqualValues(t, 2, rdb.LLen(ctx, key1).Val())
+			require.EqualValues(t, 0, rdb.Exists(ctx, key1).Val())
+			require.EqualValues(t, 2, rdb.LLen(ctx, key2).Val())
 		})
 
 		t.Run(fmt.Sprintf("BLMPOP test blocked served secondKey noCount %s", direction), func(t *testing.T) {

